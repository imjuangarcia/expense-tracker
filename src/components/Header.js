import React from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expense Tracker</h1>
    <ul>
      <li>
        <NavLink to="/dashboard" activeClassName="active">
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink to="/add" activeClassName="active">
          Add Expense
        </NavLink>
      </li>
      <li>
        <button onClick={startLogout}>Logout</button>
      </li>
    </ul>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
