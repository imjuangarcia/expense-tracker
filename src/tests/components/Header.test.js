import React from "react";
import { shallow } from "enzyme";
import Header from "../../components/Header";

test("Should render the Header component", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
