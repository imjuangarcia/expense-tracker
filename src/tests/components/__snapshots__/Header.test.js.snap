// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Header component 1`] = `
<header>
  <h1>
    Expense Tracker
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/dashboard"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/add"
      >
        Add Expense
      </NavLink>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </li>
  </ul>
</header>
`;
